setup.py

```diff
@@ -52,6 +52,7 @@
     'imagesize',
     'requests>=2.0.0',
     'typing',
+    'setuptools',
```

Comment:
:man_facepalming: please don't install setuptools from within installed version of setuptools. this causes horrible conflicts.